Sources/CvPlayer.cpp

```diff
@@ -7633,7 +7633,8 @@ int CvPlayer::getProductionNeeded(UnitTypes eUnit) const
 	iProductionNeeded *= iModifier;
 	iProductionNeeded /= 100;
 
-	if (GC.getGame().isOption(GAMEOPTION_SIZE_MATTERS))
+	// Trade units aren't impacted by SM production reduction due to ROI oddities. They can't merge anyway, so...
+	if (GC.getGame().isOption(GAMEOPTION_SIZE_MATTERS) && GC.getUnitInfo(eUnit).getUnitCombatType() != GC.getInfoTypeForString("UNITCOMBAT_TRADE"))
```

Comment:
Tested this is functional ingame, but not sure if there's a better way to do this as it's hardcoded to "UNITCOMBAT_TRADE" and seems a bit hacky? Otherwise, I think this PR is ready; changes since yesterday are mostly text, and some value edits as per discord discussion.

Reply:
> Tested this is functional ingame, but not sure if there's a better way to do this as it's hardcoded to "UNITCOMBAT_TRADE" and seems a bit hacky? Otherwise, I think this PR is ready; changes since yesterday are mostly text, and some value edits as per discord discussion.

Is there not a canMerge() function? would think there is as some units can't merge.